openedx/openedx-platform
# [ENT-503] Allow configuration of ECOMMERCE_SERVICE_WORKER_USERNAME

This pull request allows `ECOMMERCE_SERVICE_WORKER_USERNAME` to be configured in `lms.env.json`.

**JIRA tickets**: Related to [ENT-503](https://openedx.atlassian.net/browse/ENT-503)

**Dependencies**: None

**Screenshots**: N/A

**Sandbox URL**: TBD - sandbox is being provisioned.

**Merge deadline**: 16 July 2017

**Testing instructions**:

1. Pull this code.
2. In devstack, do the following:
    ```bash
    $ manage.py lms shell --settings=devstack
    >>> from django.conf import settings
    >>> settings.ECOMMERCE_SERVICE_WORKER_USERNAME
    ```
3. Observe that the response is `ecommerce_worker`.
4. Edit `lms.env.json` and add the `ECOMMERCE_SERVICE_WORKER_USERNAME` key with a distinct string value.
5. Do the following:
    ```bash
    $ manage.py lms shell --settings=devstack
    >>> from django.conf import settings
    >>> settings.ECOMMERCE_SERVICE_WORKER_USERNAME
    ```
6. Observe that the response matches the value you filled in for that variable.

**Settings**
```yaml
EDXAPP_FEATURES:
  ENABLE_COMBINED_LOGIN_REGISTRATION: true
```
